src/services/ioSignService.ts

```diff
@@ -0,0 +1,120 @@
+  withValidatedOrInternalError
+} from "../utils/responses";
+import { readableProblem } from "../../src/utils/errorsFormatter";
+import { ResponseErrorNotFound403 } from "./eucovidcertService";
+
+export default class IoSignService {
+  constructor(private readonly ioSignApiClient: ReturnType<IoSignAPIClient>) {}
+
+  /**
+   * Get the Signer id related to the user.
+   */
+  public readonly getSignerByFiscalCode = (
+    user: User
+  ): Promise<
+    | IResponseErrorInternal
+    | IResponseErrorValidation
+    | IResponseErrorNotFound
+    | IResponseErrorForbiddenNotAuthorized
+    | IResponseSuccessJson<SignerDetailView>
+  > =>
+    withCatchAsInternalError(async () => {
+      const validated = await this.ioSignApiClient.getSignerByFiscalCode({
+        body: { fiscal_code: user.fiscal_code }
+      });
+      return withValidatedOrInternalError(validated, response => {
+        switch (response.status) {
+          case 200:
+            return ResponseSuccessJson(response.value);
+          case 400:
+            return ResponseErrorValidation(
+              "Invalid request",
+              "An error occurred while validating the request body"
+            );
+          case 403:
+            return ResponseErrorNotFound403(
+              "The user associated with this profile could not be found."
+            );
+          case 500:
+            return ResponseErrorInternal(
+              `Internal server error | ${response.value}`
+            );
+          default:
+            return ResponseErrorStatusNotDefinedInSpec(response);
+        }
+      });
+    });
+
+  public readonly createFilledDocument = (
+    document_url: NonEmptyString,
+    email: EmailString,
+    family_name: NonEmptyString,
+    name: NonEmptyString,
+    signerId: Id
+  ): Promise<
+    | IResponseErrorInternal
+    | IResponseErrorValidation
+    | IResponseErrorNotFound
+    | IResponseErrorForbiddenNotAuthorized
+    | IResponseSuccessJson<FilledDocumentDetailView>
+  > =>
+    withCatchAsInternalError(async () => {
+      const validated = await this.ioSignApiClient.createFilledDocument({
+        body: {
+          document_url,
+          email,
+          family_name,
+          name
+        },
+        "x-iosign-signer-id": signerId
+      });
+      return withValidatedOrInternalError(validated, response => {
+        switch (response.status) {
+          case 201:
+            return ResponseSuccessJson(response.value);
+          case 400:
+            return ResponseErrorValidation(
+              "Invalid request",
+              "An error occurred while validating the request body"
+            );
+          case 404:
+            return ResponseErrorNotFound(
+              "Resources not found",
+              "The user associated with this profile could not be found."
+            );
+          case 500:
+            return ResponseErrorInternal(
+              readableProblem((response.value as unknown) as ProblemJson)
```

Comment:
why this cast, is it a bug in the API generator? (undefined instead of ProblemJson)

Reply:
Yes the generator does not create a ProblemJson. But I think the problem is related to the openapi structure. I will investigate

Reply:
I tried to edit the open api but it always shows as `unknown`. I think it's a generator problem

Reply:
I'd add a TODO then

Reply:
did you revert to a cast from decode?